api/src/main/java/org/openmrs/module/mirebalais/setup/ConfigurationSetup.java

```diff
@@ -0,0 +1,415 @@
+        setStatus("Migrating attachment concepts if necessary");
+        AttachmentsSetup.migrateAttachmentsConceptsIfNecessary(conceptService);
+
+        // Configure global scripts and css files defined in the configuration directory
+        setStatus("Configuring global resources");
+        GlobalResourceSetup.includeGlobalResources();
+
+        // Rebuild search index (TODO: Confirm this is intended to run here, prior to metadata loading, rather than at the end of the process(
+        if (config.shouldRebuildSearchIndex()) {
+            setStatus("Rebuilding search index");
+            Context.updateSearchIndex();
+        }
+
+        // TODO: Note here below are from MetadataSetupTask, which can be run in a separate thread or not
+
+        // Install Concepts, etc from MDS Packages
+        setStatus("Installing MDS packages");
+        MetadataSharingSetup.installMetadataSharingPackages();
+
+        // Load remaining Initializer domains that could depend on Concepts
+        setStatus("Loading initializer post-concept domains");
+        InitializerSetup.loadPostConceptDomains(config);
+
+        // Load PIH drug list
+        setStatus("Installing drug list");
+        DrugListSetup.installDrugList();
+
+        // Configure the Disposition Config based on PIH Config
+        // TODO: Note, in the various activators, this is done multiple times across many started and contextRefreshed methods
+        // TODO: Confirm that moving this here makes sense - after most metadata has been loaded
+        if (config != null && config.getDispositionConfig() != null) {
+            setStatus("Configuring disposition config");
+            dispositionService.setDispositionConfig(config.getDispositionConfig());
+        }
+
+        // Load HTML Forms  TODO: Move this over to use the initializer methods for this
+        setStatus("Loading HTML forms");
+        HtmlFormSetup.loadHtmlForms(false);
+
+        // Load Reports  TODO: Start a discussion about moving this functionality over to initializer
+        setStatus("Loading reports");
+        ReportLoader.loadReportsFromConfig();
+
+        // TODO: This depends on specific metadata to exist, so needs to go in here
+        // TODO: But we should consider if
+        //  a) this is still actually needed/used, and
+        //  b) we probably don't need the component check at all, as this is only setup if the lacolline runtime properties are found
+        //  c) if it's used, we could likely try to make it more entirely configurable from config
+        if (config.isComponentEnabled(Components.LEGACY_MPI)) {
+            setStatus("Setting up connection to Legacy MPI");
+            LegacyMasterPatientIndexSetup.setupConnectionToMasterPatientIndex(new RuntimeProperties());
+        }
+
+        // TODO: Can we just always have this scheduled task running and within the task itself, checkcheck if the component is enabled and it isn't enabled, it can just return without doing anything
```

Comment:
Sure, we could... what is the benefit of removing all the component references from here?

Reply:
Consistency mostly.  That we have a consistent backend that is always present in the same state, and it is just the metadata, other runtime configurations, and what apps/extensions that are enabled or disabled that differentiates systems.

There are very few places outside of the CALF that components are used at all, as far as I can tell.  If we can eliminate them outside of the CALF, and incorporate them into something specifically designed for use in app and extension configuration, then that might make things simpler and clearer.

Reply:
Personally I also hate the use of the OpenMRS scheduler, and always build scheduled tasks natively using Spring scheduler (see reporting and idgen).  Doing so you never run into these super annoying stack traces in the logs where various scheduled tasks defined by, say, the mirebalais module and which refer to classes in the mirebalais module fail when the mirebalais module fails to load.  The tasks just never get registered in the first place.  And modules that get removed from use take their tasks with them - there are no dangling tasks that are still in the scheduled task table that one needs to go and remember to clean up.  Just feels more robust.  The openmrs scheduler feels most appropriate to me when you want an administrator to have more control over creating, deleting, modifying, and scheduling a particular task.  But for something that is always intended to run in the background and do it's thing when necessary, I think a spring-managed task is far superior.